### PR TITLE
Update README.md and split into several related files

### PR DESCRIPTION
**Pull request by ThanGerlek**
A PR for #451. Adds, updates, clarifies, and reorganizes "getting started" info for new devs joining the autograder team.

- Extract getting started info from `README.md`
- Reorganize information by use case (e.g. isolate Windows-specific info)
- Remove/replace outdated information
- Add relevant "getting started" info
- General improvements to setup instructions

### CODE REVIEW

**Comment by ThanGerlek** [comment]
I removed as much incorrect/outdated info as I could, but there were several things I wasn't sure about, and I'm sure I missed some. Keep an eye out for anything I mistakenly changed

**Comment by frozenfrank** [comment]
Thanks @ThanGerlek! I'll review this when I have a moment.

Quickly though, you may noticed that I added a new line to the end of your PR description. It contains a special [closing keyword](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue) which GitHub recognizes to directly link this PR to the issue you opened earlier (great job BTW). These closing keywords only work when placed inside the PR description (putting it in a comment won't do the trick).

You'll also notice that I added a link to the issue in #444 which is the parent issue of these kinds of changes. Previously, this kind of adjustment had been reference by text only, but I hadn't yet created an issue for it. Thank you for doing so!

**Comment by frozenfrank** [comment]
Upon closer examination of the sources, I understood why you had not wanted this to **close** the #451 issue: You intend for future work to actually clean up the content whereas this PR mainly affects the organization of the files.

I removed the closing keywords from the PR description accordingly.

File: getting-started/getting-started.md
Comment (frozenfrank) [inline getting-started/getting-started.md:7]: There is currently no need for special instruction pages for Linux and MacOS devices. According to the [YAGNI](https://martinfowler.com/bliki/Yagni.html), we should not include those pages. We don't even expect to need them!

We could replace this section with the following Windows specific message (which displays as an appropriate warning box for users).

```md
> [!WARNING]
> The autograder unfortunately requires special setup instructions
> in order to operate on **Windows machines**.
> 
> Read [Getting Started on Windows](windows.md) before continuing.
```

File: README.md
Comment (frozenfrank) [inline README.md:63]: We can provide a link to a filtered view of the issues page which only displays those with the referenced tag.

Let's add this link over the words "good first issue": 
https://github.com/softwareconstruction240/autograder/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22

File: getting-started/windows.md
Comment (frozenfrank) [inline getting-started/windows.md:41]: See earlier comment about removing recommendations to setup with VSCode.

**Review by frozenfrank — CHANGES_REQUESTED** [review]
This structural change looks great, @ThanGerlek, and I really appreciate the expanded and clarified content along the way!

I've provided several comments in files and lines of code with ways we may be able to improve the presentation of information.

After working through all the content, I realized that I ended up proposing two opposing ways of dealing with the platform-specific setup instructions. Both sets of comments remain in the review, but I'll clarify which I recommend. The opposing paths are:
1. Remove unused/unneeded setup instruction files (recommended)
2. Change content of filler files (**not** recommended)

File: README.md
Comment (frozenfrank) [inline README.md:51]: Since this information is already contained in the `getting-started/getting-started.md` file, and because the information is more appropriately contained in that file, I recommend removing it entirely from this root `README.md` file.

Subsequent changes to the wording of the next sentence would also be appropriate.

File: getting-started/getting-started.md
Comment (ThanGerlek) [reply getting-started/getting-started.md:85]: I don't see this... What proposed change are you referring to?

File: getting-started/getting-started.md
Comment (ThanGerlek) [reply getting-started/getting-started.md:7]: Agreed, but I left them in case @19mdavenport or any of the other TAs have useful information to add. I only know about Windows 🤷‍♂️

File: README.md
Comment (ThanGerlek) [reply README.md:55]: Eh, I was tryna be clever. It's changed.

File: getting-started/windows.md
Comment (ThanGerlek) [reply getting-started/windows.md:5]: I've added some introductory info. Do you think it's clear?

File: getting-started/db-insert-statements/insert-config-database.md
Comment (frozenfrank) [reply getting-started/db-insert-statements/insert-config-database.md:1]: Those are important things to consider. Given that additional context, recognizing that new users are only expected to use one of them, I think it's fine to leave "as is" until we reorganize the insert statement organizations.

File: getting-started/getting-started.md
Comment (frozenfrank) [reply getting-started/getting-started.md:85]: I added the word "USER" after "DELETE" to be parallel to "CREATE USER."

I don't actually know which technical entities are being referred to in the statement though, and I could be wrong.

I searched online for SQL statements relating to CREATE/DELETE user, but I found `DROP LOGIN` and `DROP USER` instead. Can we get a link to the "administrator privileges documentation that describes which restricted privileges are required?

File: getting-started/windows.md
Comment (frozenfrank) [inline getting-started/windows.md:4]: Thank you for these clarifications!

File: getting-started/windows.md
Comment (frozenfrank) [inline getting-started/windows.md:51]: I'll convert these to a special GitHub alert callout.

**Review by frozenfrank — COMMENTED** [review]
These changes are looking great!

There's a presentation change I want to make; but otherwise, a big thank you for making these changes happen!

File: getting-started/getting-started.md
Comment (ThanGerlek) [reply getting-started/getting-started.md:86]: Right. I'm blind.

"CREATE USER" and "DELETE" are just the exact names of the checkboxes I needed to click on in MySQL Workbench (under Users and Privileges/Administrative Roles/Global Privileges) in order to make it work. I have no idea what MySQL codes they map to. But that was last year, and some experimentation today revealed you need pretty much everything anyway (it needs to be able to call GRANT ALL on the student user, so it needs at least "GRANT ALL"-level privileges). So listing them here really isn't useful anymore.

**Review by frozenfrank — APPROVED** [review]
This looks good to me, but I would wait for a review by @19mdavenport before merging it.

What else would be needed in order to actually close #451? (Have we made the adjustments already; if not, can you specifically update #451 to indicate what work remains?)

**Comment by frozenfrank** [comment]
@ThanGerlek FYI It seems that we keep fighting over trailing whitespace on the lines.

Can you make sure you have these "On Save" settings enabled?
* Remove trailing whitespace
* Remove trailing blank lines at the end of saved files
* Ensure every saved file ends with a line break

image

**Comment by ThanGerlek** [comment]
Done. Thanks for catching that. Maybe a code style XML or something would be useful for the autograder as well as the chess project... not exactly top priority though

**Review by frozenfrank — COMMENTED** [review]
OK, now I'm fully comfortable with the presentation of the content. I feel like the information is a readable as it can be, and I'm happy with the way it presents on the screen.

I did chat with @19mdavenport this morning, and he mentioned that there may be some content adjustments we need to make. Let's wait to hear back from him— he's a busy man, but he'll look at this soon enough.

**Comment by webecke** [comment]
Just a small thing, GitHub treats the `docs/` folder specially, especially in a way that will help with work going on in #448. Should we put all these files in a `docs/` folder or do we specifically want it to be in a separate `getting-started/` folder?

**Comment by frozenfrank** [comment]
> Just a small thing, GitHub treats the `docs/` folder specially, especially in a way that will help with work going on in #448. Should we put all these files in a `docs/` folder or do we specifically want it to be in a separate `getting-started/` folder?

Great suggestion, @webecke! This is accounted for in 571cca5de05fea11072bc78b8ccf9ee8c1cba618.

**Comment by frozenfrank** [comment]
Also marking this as resolving #335 since the clarifications and adjustments to the language indicate no further requirement to have Canvas API tokens.

**Comment by webecke** [comment]
I consolidated the development and getting started sections of the README.md into one small section and moved it to the top. If we're opposed to that we can change it back, but for almost everyone coming to look at the Autograder repo the info they need is in the Getting Started Guide and Contribution Guide. Thus I propose they should be at the top.

**Comment by ThanGerlek** [comment]
Updated the PR description to represent recent changes.

At this point this PR will probably close issue #452, although not until we're all happy with it, so I've changed it to a draft PR as well.

**Comment by frozenfrank** [comment]
@19mdavenport I think you're review is the last thing we're looking for on this PR.

@ThanGerlek Are you aware of any other changes we're still waiting for?

**Comment by ThanGerlek** [comment]
Nope, that's all. Not just Michael's review but also any additions he thinks might be useful

**Comment by frozenfrank** [comment]
> Nope, that's all. Not just Michael's review but also any additions he thinks might be useful

I went ahead and marked it as "ready to review" since you have contributed everything you are planning on for this PR. Other reviewers and contributors may weigh in, but now is the time for that process to be happening. (Typically, the "draft" stage is a signal that other people should _not_ look at this PR because the PR author is still working on critical aspects of it.)

**Comment by webecke** [comment]
I vote we go ahead and merge this in if no one else has things to add. @frozenfrank @ThanGerlek @19mdavenport

**Pull request merged**
